src/Microsoft.Extensions.DependencyModel/DependencyContextLoader.cs

```diff
@@ -16,26 +16,26 @@ public class DependencyContextLoader
         private readonly string _entryPointDepsLocation;
         private readonly string _runtimeDepsLocation;
         private readonly IFileSystem _fileSystem;
-        private readonly IDependencyContextReader _jsonReader;
+        private readonly Func<IDependencyContextReader> _jsonReaderFactory;
 
         public DependencyContextLoader() : this(
             DependencyContextPaths.Current.Application,
             DependencyContextPaths.Current.SharedRuntime,
             FileSystemWrapper.Default,
-            new DependencyContextJsonReader())
+            () => new DependencyContextJsonReader())
         {
         }
 
         internal DependencyContextLoader(
```

Comment:
Is this constructor used anywhere else? Can it be made 'private' or even removed?